### ISSUE TITLE
gpt2 finetuning on tpu

### ISSUE THREAD

**Issue by kbrajwani** [alexorona/lm_tf_trainer#1]
hey @alexorona i have tried to implement this script in following notebook. But as i am no aware of finetuning of gpt i can't measure that its going good. so can you please help me too check that model is learning something or i am doing any mistakes.
https://colab.research.google.com/drive/1HBCUOZC4vejyAzb8Gq8MqOVWSxu7Ux5o?usp=sharing

**Comment by alexorona** [comment]
@kbrajwani The loss reported by the temporary trainer isn't quite right. I made an adjustment which should provide a more accurate figure. There's also a print-out. As for your notebook, I'm happy to report that you are definitely training. Without changing your upstream code, I added this to the end of the notebook and received a coherent response:

```
from transformers import GPT2TokenizerFast
tokenizer = GPT2TokenizerFast.from_pretrained('gpt2-medium')

# Model input
text = "First Citizen: Before we proceed any further, hear me speak. All: Speak, speak"

# Format and generate
inputs = tokenizer.encode_plus(text, return_tensors = 'tf')
output = model.generate(inputs['input_ids'], 
                        attention_mask = inputs['attention_mask'],
                        max_length = 100)
np_output = output.numpy()[0].tolist()
tokenizer.decode(np_output)
-----
Setting `pad_token_id` to 50256 (first `eos_token_id`) to generate sequence

'First Citizen: Before we proceed any further, hear me speak. All: Speak, speak.

First Citizen:
I the great toe! why the great toe?

MENENIUS:
For that, being one o' the lowest, basest, poorest,
Of this most wise rebellion, thou go'st foremost:
Thou rascal, that art worst in blood to run,
Lead'st first to win some vantage.
But make you ready'

```

**Comment by kbrajwani** [comment]
Thanks @alexorona i am happy to hear that at-least its training.
